lib/client/sse_client.dart

```diff
@@ -36,7 +40,25 @@ class SseClient extends StreamChannelMixin<String> {
         .listen(_onOutgoingMessage, onDone: _onOutgoingDone);
     _eventSource.addEventListener('message', _onIncomingMessage);
     _eventSource.addEventListener('control', _onIncomingControlMessage);
-    _eventSource.onError.listen(_incomingController.addError);
+    _eventSource.onOpen.listen((_) {
+      _errorTimer?.cancel();
+    });
+    _eventSource.onError.listen((error) {
+      if (!withRetry) {
+        _incomingController.addError(error);
+        _eventSource.close();
+      } else {
+        if (!(_errorTimer?.isActive ?? false)) {
+          // By default the SSE client uses keep-alive.
+          // Allow for a retry to connect before giving up.
+          _errorTimer = Timer(const Duration(seconds: 5), () {
```

Comment:
Where did 5 seconds come from?

Reply:
Just a number that has worked well in practice for internal systems. Realistically the event source reconnect should be sub second. This gives plenty of overhead.